framework/src/controller/schema/application.js

```diff
@@ -13,6 +13,211 @@ module.exports = {
 	constants: {
 		id: '#/app/constants',
 		type: 'object',
+		required: [
+			'ACTIVE_DELEGATES',
+			'BLOCK_SLOT_WINDOW',
+			'ADDITIONAL_DATA',
+			'BLOCK_RECEIPT_TIMEOUT',
+			'EPOCH_TIME',
+			'FEES',
+			'MAX_PAYLOAD_LENGTH',
+			'MAX_PEERS',
+			'MAX_SHARED_TRANSACTIONS',
+			'MAX_TRANSACTIONS_PER_BLOCK',
+			'MAX_VOTES_PER_TRANSACTION',
+			'MAX_VOTES_PER_ACCOUNT',
+			'MIN_BROADHASH_CONSENSUS',
+			'MULTISIG_CONSTRAINTS',
+			'NETHASHES',
+			'NORMALIZER',
+			'REWARDS',
+			'TOTAL_AMOUNT',
+			'UNCONFIRMED_TRANSACTION_TIMEOUT',
+			'EXPIRY_INTERVAL',
+		],
+		properties: {
+			ACTIVE_DELEGATES: {
+				type: 'integer',
+				min: 0,
+			},
+			BLOCK_SLOT_WINDOW: {
+				type: 'integer',
+			},
+			ADDITIONAL_DATA: {
+				type: 'object',
+				required: ['MIN_LENGTH', 'MAX_LENGTH'],
+				properties: {
+					MIN_LENGTH: {
+						type: 'integer',
+					},
+					MAX_LENGTH: {
+						type: 'integer',
+					},
+				},
+			},
+			BLOCK_RECEIPT_TIMEOUT: {
+				type: 'integer',
+			},
+			EPOCH_TIME: {
+				type: 'object',
+			},
+			FEES: {
+				type: 'object',
+				$ref: '#/fees',
+			},
+			MAX_PAYLOAD_LENGTH: {
+				type: 'integer',
```

Comment:
There must be a minimum value for it. Please refer to https://lisk.io/documentation/lisk-protocol/transactions to calculate what the minimum a single transactions can have.

Reply:
Can reference value within the same schema with $data property, but not outside in a higher scope.

Reply:
Decided to not verify this for now as it is subject to change.